### PR TITLE
fixes for bug Major 1

### PR DESCRIPTION
**Pull request by idirall22**

### CODE REVIEW

File: contracts/NodeOperatorRegistry.sol [hunk 1]
```diff
@@ -1074,6 +1104,22 @@ contract NodeOperatorRegistry is
                 index++;
             }
         }
+
+        // Bug: Major 1.3 and 1.4
+        // If (index != totalActiveNodeOperator - 1) that mean there are empty fields
+        // in the array, it happens only in the case when a validator was unstaked
+        // directly from the stakeManager and the node operator registry not yet updated
+        // the operator status.
+        if (index != totalActiveNodeOperator - 1) {
```
Comment (GitMark0) [inline contracts/NodeOperatorRegistry.sol:1113]: I'm going to need some clarification. 🤔

File: contracts/StMATIC.sol [hunk 2]
```diff
@@ -373,7 +373,12 @@ contract StMATIC is
             .getOperatorInfos(true);
 
         for (uint256 i = 0; i < operatorInfos.length; i++) {
-            IValidatorShare(operatorInfos[i].validatorShare).withdrawRewards();
+            // Bug: Major 1.3
+            IValidatorShare vs = IValidatorShare(operatorInfos[i].validatorShare);
```
Comment (GitMark0) [inline contracts/StMATIC.sol:377]: This is good, but I've already fixed this one. 😸 
https://github.com/Shard-Labs/PoLido/blob/fix/post-audit/contracts/StMATIC.sol#L379

**Review by GitMark0 — CHANGES_REQUESTED** [review]

File: contracts/NodeOperatorRegistry.sol [hunk 3]
```diff
@@ -1057,11 +1079,19 @@ contract NodeOperatorRegistry is
...
             NodeOperator storage no = operators[operatorId];
 
-            if (no.status == NodeOperatorStatus.ACTIVE) {
+            // Bug: Major 1.2
+            IStakeManager.Validator memory validator = sm.validators(
+                no.validatorId
+            );
+            if (
+                no.status == NodeOperatorStatus.ACTIVE &&
+                validator.status == IStakeManager.Status.Active
```
[the remark this answers is not in the document]
Comment (idirall22) [reply contracts/NodeOperatorRegistry.sol:1093]: The second one is enough

File: contracts/NodeOperatorRegistry.sol (hunk 1 above)
Comment (idirall22) [reply contracts/NodeOperatorRegistry.sol:1113]: Yes
Let's say we have 3 validators, then 1 validator was unstaked by the stakemanager.
In this case the nodeOperator contract we still have 3 active validator. SO wen we call getOperatorInfo we are creating an array of 3 validator. Then when we use this check (validator.status == IStakeManager.Status.Active) we are going to add only 2 validators to this array and the third one will be empty.
So, later this empty operator will revert when delegate, or distribute rewards.
This is why I added this additional loop to remove the empty operator.
Is that make sense for you?

**Pull request closed without merging**